pkg/network/namespace/monitor.go

```diff
@@ -0,0 +1,25 @@
+package namespace
+
+import (
+	"context"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+)
+
+// Monitor the dmz namespace for updates
+func Monitor(ctx context.Context, name string) (chan netlink.AddrUpdate, error) {
+	ns, err := netns.GetFromName(name)
+	if err != nil {
+		return nil, err
+	}
+
+	ch := make(chan netlink.AddrUpdate)
+	if err := netlink.AddrSubscribeAt(ns, ch, ctx.Done()); err != nil {
+		close(ch)
+		ns.Close()
```

Comment:
are we sure this will always be called ? otherwise we could leak a file descriptor here.

Reply:
I checked the AddrSubscribe code, and they close the namespace once the ctx channel is done. Hence it's safe to not close it as long as the subscribe has been successful.